7217 add all compatible android build

aimee-mcneil-melville opened the pull request:
<!-- IMPORTANT!
  - Every PR must reference an issue; this helps to explain the intent of the PR
 -->

Fixes #7217

# 👩🏻‍💻 What does this PR do?

<!-- Explain the changes you made -->

Add a universal android apk using armeabi-v7a and include it in the github action

The android build action still runs on push of a v* tag, but now uploads two artifacts:
- Arm64Build (64bit only)
- Universal_Build (runs on 32bit and 64bit devices) 

These artifacts can be downloaded separately. Due to github behaviour artifacts can only be uploaded to github as zip files - therefore they must be downloaded and unzipped before being used. I did remove the surrounding folder so each zip only contains the apk - raised in #6322. 

Before I could start with the build, I needed to remove Extism. This was also being done at the same time here: #7265 
I matched my changes to this PR, the only differences now are in the cargo.lock. There are a few commits around it though

Arm64Build - tested on 64bit android tablet
Universal_Build - tested on 64bit android tablet and two 32bit android phones. Screenshots are of the fridge tag import test on each phone (thanks @GeronimoJohn!)
![2025-04-14 17 48 33 Medium](https://github.com/user-attachments/assets/df2707c5-6f62-462a-80ee-b1ac8d05adad)

![2025-04-14 17 48 39 Medium](https://github.com/user-attachments/assets/57bf76fa-8a13-4372-a128-d50d789f9fa8)


<!-- why are the changes needed -->

<!-- Add a screenshot if there are UI changes  -->

## 💌 Any notes for the reviewer?

The app seems to often not open properly the first time after installing on the android phones I tested on. Close -> reopen and it works. This didn't seem to be an issue with the same apk on a tablet

There is an action available to automatically add the apks to releases (this gets around the zipping, see #6322 comments). 
We could add this for release versions (this github action), and create a new action for test apks. This testing action could have no automated release upload, but a different condition for the action starting (and maybe use the android:build:debug command)

The action now takes around 20 minutes - increase of about 5-7 minutes

<!-- Do you have any specific questions for the reviewer? -->

<!-- Is there a high risk/complicated change they should focus on? -->

<!-- any general areas of the codebase touched? any side effects caused? -->

<!-- Anything half cooked but going to be finished off in a different PR? -->

# 🧪 Testing

<!-- Explain the steps you'd take to test the changes of this PR manually -->

- [ ] Test reports still work as per #7265 
- [ ] Push a tag beginning with v - this will start a build in Github actions
- [ ] Log into github on the device you wish to test
- [ ] Download the artifact needed for the device
- [ ] Unzip, then install the app
- [ ] Test that it works! Running on a 32bit device should see cold chain section - not the full app

# 📃 Documentation

- [ ] **Part of an epic**: documentation will be completed for the feature as a whole
- [ ] **No documentation required**: no user facing changes or a bug fix which isn't a change in behaviour
- [x] **These areas should be updated or checked**: <!-- _(e.g.)_ New `issued` column in `Requisitions` indicates stock quantity already in shipments -->
  1. Unsure - do we need documentation on when to use which apk?
 


# 📃 Reviewer Checklist

The PR Reviewer(s) should fill out this section before approving the PR

**Breaking Changes**
- [ ] No Breaking Changes in the Graphql API
- [ ] Technically some Breaking Changes but not expected to impact any integrations

**Issue Review**
- [ ] All requirements in original issue have been covered
- [ ] A follow up issue(s) have been created to cover additional requirements

**Tests Pass**
- [ ] Postgres
- [ ] SQLite
- [ ] Frontend

aimee-mcneil-melville commented:
Have added a readme update here #7336 

I ran yarn android:build:debug to test it - this also now builds both apks

andreievg commented on client/packages/android/package.json:
```diff
@@ -18,7 +18,7 @@
     "build:server": "./build_remote_server_libs.sh",
     "build:debug": "yarn build:server && ./gradlew assembleDebug",
-    "build:release": "yarn build:server && ./gradlew assembleRelease",
+    "build:release": "yarn build:server && ./gradlew assembleArm64Release assembleUniversalRelease",
```
Wow this is super cool. Impressive.
Since there is no direct way to search for assembleArm64Release globally (and the other one), can you please add a comment: "//": "assembleArm64Release and assembleUniversalRelease refers to arm64 and universal in client/packages/android/app/build.gradle"

andreievg reviewed (commented):
Sorry forgot something, can we have the ability to build debug version 'quickly' without compiling for both architectures ? This is only to do locally, it should be pretty simple ? Doesn't block this PR, it's ok to make another issue

aimee-mcneil-melville commented:
Thanks for the review @andreievg! 

I found that specifying ```assembleArm64Release assembleUniversalRelease``` in the command wasn't necessary - ```assembleRelease``` runs both. I updated this and added a comment.

Issue for new scripts created here: #7376
